pywr/_core.pyx

```diff
@@ -754,14 +779,42 @@ cdef class AggregatedNode(AbstractNode):
             return self._max_flow
         return self._max_flow_param.get_value(scenario_index)
 
+    cpdef double[:] get_factors(self, ScenarioIndex scenario_index):
+        """Get node factor for a give timestep
+        """
+        if  self._factor_parameters is None:
+            return self.factors
+        else:
+            return np.array([p.get_value(scenario_index) for p in self.factor_parameters], np.float64)
+
+    cpdef double[:] get_factors_norm(self, ScenarioIndex scenario_index):
```

Comment:
This could be made more efficient if an "out" array was an optional argument. Instead of creating a new array the function would write to the provided array instead.

Reply:
Is it possible to have an optional arg for a cpdef function. From a quick test, I was unable to get this to work.

Reply:
Yes, see `_reset_storage_only` in this file. You need to declare it in the pxd.

https://cython.readthedocs.io/en/latest/src/userguide/language_basics.html#optional-arguments